Lib/urllib/parse.py

```diff
@@ -480,6 +480,29 @@ def urlsplit(url, scheme='', allow_fragments=True):
     v = SplitResult(scheme, netloc, url, query, fragment)
     return _coerce_result(v)
 
+# typed=True avoids BytesWarnings being emitted during cache key
+# comparison since this API supports both bytes and str input.
+@functools.lru_cache(typed=True)
+def pathsplit(path):
```

Comment:
while you've got a use for this, i'm wary about adding yet another public API to urllib.parse code.

Reply:
Hmm, your previous comment suggested you wanted it in `urllib.parse`.  I'm wary of adding a new API as well. Now, I'm not even totally sure that having `http.server` use `pathsplit()` is correct at all.  RFC 2616 says:

> The most common form of Request-URI is that used to identify a
   resource on an origin server or gateway. In this case the absolute
   path of the URI MUST be transmitted (see [section 3.2.1](https://www.rfc-editor.org/rfc/rfc2616#section-3.2.1), abs_path) as
   the Request-URI, and the network location of the URI (authority) MUST
   be transmitted in a Host header field.

My current understanding is that `http.server` implements HTTP 1.0 and so it doesn't handle an absolute URI as Request-URI.  In that case, I think having `pathsplit()` is of pretty limited use since it probably is only useful if you are parsing an HTTP 1.0 request line.

Reply:
I've removed `pathsplit()` as a public API.

Reply:
The http1.1 rfc 2616 says it MUST be abs_path, not absoluteURI. Thus what we handle is correct (and we do have http/1.1 support IIRC).  If you follow the modern RFC 7231 i believe you end up in https://datatracker.ietf.org/doc/html/rfc3986#section-3.3